src/FEValues/GeometryMapping.jl

```diff
@@ -134,15 +134,17 @@ end
     return MappingValues(fecv_J, nothing)
 end
 
-#= @inline function calculate_mapping(geo_mapping::GeometryMapping{2}, q_point, x)
+@inline function calculate_mapping(geo_mapping::GeometryMapping{2}, q_point, x)
     J = zero(otimes_returntype(eltype(x), eltype(geo_mapping.dMdξ)))
+    sdim, rdim = size(J)
+    (rdim != sdim) && error("hessian for embedded elements not implemented (rdim=$rdim, sdim=$sdim)")
```

Comment:
Are there any costs associated with these checks (and in `apply_mapping!`), or are they inferred by the compiler?

Reply:
I guess this is a possibility but I am not sure how to check that. Do you want to remove it and let the it error further down in the function?